template/koa-rest-api/src/tracing.ts

```diff
@@ -0,0 +1,49 @@
+/**
+ * Opentelemetry tracing initialisation. This is a standalone ts/js module that is not
+ * referenced by application source code directly. It is required at runtime using the
+ * node command's `--require` argument, see Dockerfile for details.
+ */
+
+import { propagation } from '@opentelemetry/api';
+import { CompositePropagator } from '@opentelemetry/core';
+import { CollectorTraceExporter } from '@opentelemetry/exporter-collector-grpc';
+import { AwsInstrumentation } from '@opentelemetry/instrumentation-aws-sdk';
+import { HttpInstrumentation } from '@opentelemetry/instrumentation-http';
+import { B3InjectEncoding, B3Propagator } from '@opentelemetry/propagator-b3';
+import { NodeSDK } from '@opentelemetry/sdk-node';
+
+const main = async () => {
```

Comment:
Does `async` code loaded via `--require` behave any differently to `async` code called from the entry point? ie. is there a race here between instrumentation being initialised and the first requests being handed by the service?

Reply:
Good point. I think it's definitely a possibility that promises aren't fulfilled before the app loads. Today there's no way to pause for async execution in the bootstrap file supplied via `--require x.js`.

Really depends on what happens inside of the `sdk.start` fn: https://www.npmjs.com/package/@opentelemetry/sdk-node. Looking in code it only really returns a pending promise when we want auto resource detection which is off: https://github.com/seek-oss/skuba/pull/706/files#diff-4836bf38ef9493029c23deb46fdf06733bbc8b8cec520b9f538ea1a68cebf8eaR33

I'm going to remove async from the fn to keep it simple. However it's only hiding the possible problem with pending promise from `sdk.start()`